packages/openapi-framework/index.ts

```diff
@@ -259,7 +259,19 @@ export default class OpenAPIFramework implements IOpenAPIFramework {
               if (operationId && operationId in this.operations) {
                 const operation = this.operations[operationId];
                 acc[METHOD_ALIASES[method]] = (() => {
-                  const innerFunction: any = operation;
+                  /**
+                   * We have two options:
+                   *
+                   * 1. the middleware gets bound + dependency injected, this may be breaking.
+                   * 2. we pick the last middleware as the operation handler. This means we cannot support
+                   *    _after_ middlewares (though not a common express pattern)
+                   */
+                  const innerFunction: any = Array.isArray(operation)
+                    ? operation.map((middleware) =>
+                        middleware.bind({ dependencies: this.dependencies })
+                      )
+                    : operation.bind({ dependencies: this.dependencies });
+
                   innerFunction.apiDoc = methodDoc;
```

Comment:
i see, so you're just creating a copy of all the functions so that when apiDoc is assigned as a property it will be on the new copy not on the reference.  i think the implementation you have here is fine.  thank you!